spec/rspec/rails/matchers/have_reported_error_spec.rb

```diff
@@ -0,0 +1,233 @@
+require "rspec/rails/matchers/have_reported_error"
+
+RSpec.describe "have_reported_error matcher" do
+  class TestError < StandardError; end
+  class AnotherTestError < StandardError; end
+
+  let(:mock_error_reporter) { double("ErrorReporter") }
+  let(:subscribers) { [] }
+
+  before do
+    allow(Rails).to receive(:error).and_return(mock_error_reporter)
+
+    allow(mock_error_reporter).to receive(:subscribe) do |subscriber|
+      subscribers << subscriber
+    end
+
+    allow(mock_error_reporter).to receive(:unsubscribe) do |subscriber|
+      subscribers.delete(subscriber)
+    end
+
+    allow(mock_error_reporter).to receive(:report) do |error, **attrs|
+      subscribers.each { |subscriber| subscriber.report(error, **attrs) }
+    end
+  end
+
+  after do
+    subscribers.clear
+  end
+
+  describe "basic functionality" do
+    it "passes when an error is reported" do
+      test_block = proc do
+        Rails.error.report(StandardError.new("test error"))
+      end
+
+      expect(test_block).to have_reported_error
```

Comment:
This suspiciously looks like passing an argument, and like the “value expectations” syntax.

Do you think a proper “this matcher doesn’t work with value expectations “ error will be shown to users on an attempt to use it mistakenly?